btree/btree.c

```diff
@@ -4237,37 +4134,38 @@ int64_t btree_destroy_tree_tick(struct m0_sm_op *smop)
 
 	M0_PRE(bop->bo_op.o_sm.sm_state == P_INIT);
 	M0_PRE(bop->bo_arbor != NULL);
-	M0_PRE(bop->bo_arbor->t_desc != NULL);
-	M0_PRE(node_invariant(bop->bo_arbor->t_desc->t_root));
-
-	/** The following pre-condition is currently a
-	 *  compulsion as the delete routine has not been
-	 *  implemented yet.
-	 *  Once it is implemented, this pre-condition can be
-	 *  modified to compulsorily remove the records and get
-	 *  the node count to 0.
-	 */
-	M0_PRE(node_count(bop->bo_arbor->t_desc->t_root) == 0);
 
-	tree_put(bop->bo_arbor->t_desc);
-	/**
-	 * TODO: Currently just deleting the tree root node, as the assumption
-	 * is tree will only have root node at this point.
-	 */
-	m0_rwlock_write_lock(&list_lock);
-	ndlist_tlink_del_fini(bop->bo_arbor->t_desc->t_root);
-	m0_rwlock_write_unlock(&list_lock);
+	if (bop->bo_arbor->t_desc != NULL &&  bop->bo_arbor->t_desc->t_ref > 0) {
+		M0_PRE(node_invariant(bop->bo_arbor->t_desc->t_root));
 
-	/**
-	 * ToDo: We need to capture the changes occuring in the
-	 * root node after tree_descriptor has been freed using
-	 * m0_be_tx_capture().
-	 * Only those fields that have changed need to be
-	 * updated.
-	 */
+		/** The following pre-condition is currently a
+		 *  compulsion as the delete routine has not been
+		 *  implemented yet.
+		 *  Once it is implemented, this pre-condition can be
+		 *  modified to compulsorily remove the records and get
+		 *  the node count to 0.
+		 */
```

Comment:
make changes as described in comments as delete routine has been implemented and remove the comments.

Reply:
Oh, is it? I need to check this.

Reply:
On discussion, it is a prerequisite that all records should be deleted before reaching destroy.
Hence, there is no need to add the delete routine.
Comment has been removed.